### DISCUSSION TITLE
Use prisma generated non-nullable type in client code, where null would be useful (e.g. initializing useState())

### DISCUSSION THREAD

**Discussion by michael-mcgreal** [prisma/prisma#15303] [Q&A] [answered]
I often want to use the type of a model in client code for reasons other than database operations. And the imported types from @prisma/client are usually non-nullable.

So if I'm applying this type it to something like `useState<PrismaType>(null)` – that requires an initial value, and where null would be best – null is not allowed.

Some solutions I've tried:
- using placeholder values (such as an empty string)
- defining a new union type that uses the prisma/client type | null
  - manually managing types like this seems like an anti-pattern, but it is better than allowing nulls in the prisma schema just for a useState initial value!

Does anyone have better recommendations? Thank you.

**Comment by austincrim** [comment, marked as answer]
Hey @mcgrealife,

I think defining a union type based on the Prisma type is the way I would go. It's not possible to customize the generated Prisma types, and it should be relatively low maintenance 👍.

**Reply by michael-mcgreal** [reply]
@austincrim Thanks for confirming this best practice! It is lower maintenance than expected. Much appreciated 🙏
